### PR TITLE
Add Math.cs performance tests.

### PR DESCRIPTION
**Pull request by KostaVlev**
Add Math.cs performance tests. (#18765)

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/Performance/System/Perf.Math.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        [Benchmark]
+        public void Floor_Decimal()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < 3000000; i++)
+                    {
+                        Math.Floor(45.2m);
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/Performance/System/Perf.Math.cs]: Note that the Math operations are often treated as intrinsics by the codegenerators. If you call the Math operation with a constant, the codegenerator may just replace it with a constant - without actually calling the method.

File: src/System.Runtime.Extensions/tests/Performance/System/Perf.Math.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        [Benchmark]
+        public void Log_DoubleDouble()
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < 6000000; i++)
+                    {
+                        Math.Log(10, 5);
```
Comment (jkotas) [inline src/System.Runtime.Extensions/tests/Performance/System/Perf.Math.cs]: The math operations on floating point numbers are often treated as inrinsics by the code generators. If you call them on a constant, the codegenerator may replace the whole expression by a constant - without actually calling the function.

**Comment by tannergooding** [comment]
It may also be worth noting we have performance tests for most of `System.Math` and `System.MathF` in CoreCLR: https://github.com/dotnet/coreclr/tree/master/tests/src/JIT/Performance/CodeQuality/Math/Functions

**Comment by KostaVlev** [comment]
I am closing this PR because I don't see what else is worth to test since there are already tests for Math.cs.

**Pull request closed without merging**